src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -333,6 +339,9 @@ def set_assessment_substatus_json(self, status=Constants.STATUS_TRANSITIONING, c
         # Wrap assessment summary into assessment substatus
         self.__assessment_substatus_json = self.__new_substatus_json_for_operation(Constants.PATCH_ASSESSMENT_SUMMARY, status, code, json.dumps(self.__assessment_summary_json))
 
+        # Set force truncation true when final status is success or error
+        self.__set_force_truncation_on_terminal_status(substatus_status=status)
```

Comment:
Point to note: Force truncation is being set on the basis of a single substatus's status. It is not checking/looking for terminal status within all substatuses's

For eg: at this point in code, PatchAssessmentSummary has a terminal status but PatchInstallationSummary is still transitioning. This code will apply truncation.

Noting it down to ensure this is the desired behavior.

Reply:
<img width="712" alt="image" src="https://github.com/Azure/LinuxPatchExtension/assets/127874208/cf7058a8-91d2-434a-b87a-5cbfc1a5a869">